src/ape_test/accounts.py

```diff
@@ -6,36 +6,49 @@
 
 from ape.api import TestAccountAPI, TestAccountContainerAPI, TransactionAPI
 from ape.types import AddressType, MessageSignature, TransactionSignature
-from ape.utils import GeneratedDevAccount, cached_property, generate_dev_accounts
+from ape.utils import generate_dev_accounts
 
 
 class TestAccountContainer(TestAccountContainerAPI):
+    _accounts: List
+
+    def __init__(self, **kwargs):
+        generated_accounts = generate_dev_accounts(
+            self.config["mnemonic"], number_of_accounts=self.config["number_of_accounts"]
+        )
+        test_accounts = [
+            TestAccount(index=i, address_str=acc.address, private_key=acc.private_key)
+            for i, acc in enumerate(generated_accounts)
+        ]
+        super().__init__(**kwargs)
+        self._accounts = test_accounts
+
     @property
     def config(self):
         return self.config_manager.get_config("test")
 
-    @cached_property
-    def _dev_accounts(self) -> List[GeneratedDevAccount]:
-        mnemonic = self.config["mnemonic"]
-        return generate_dev_accounts(mnemonic, number_of_accounts=self.config["number_of_accounts"])
-
     @property
     def aliases(self) -> Iterator[str]:
         for index in range(len(self)):
             yield f"dev_{index}"
 
     @property
     def accounts(self) -> Iterator["TestAccount"]:
-        for index in range(len(self)):
-            account = self._dev_accounts[index]
-            yield TestAccount(
-                index=index,
-                address_str=account.address,
-                private_key=account.private_key,
-            )
+        for account in self._accounts:
+            yield account
 
     def __len__(self) -> int:
-        return len(self._dev_accounts)
+        return len(self._accounts)
+
+    def add_account(self) -> "TestAccount":
```

Comment:
I think the function would be better named `create_account`, since we're not adding an already existing account to the container, we're creating a new one.

Reply:
ok